web/src/stream_popover.js

```diff
@@ -446,6 +447,12 @@ export async function build_move_topic_to_stream_popover(
                 send_notification_to_new_thread,
                 send_notification_to_old_thread,
                 propagate_mode,
+                propagate_mode === "change_one"
+                    ? {
+                          new_stream_id: select_stream_id || current_stream_id,
+                          new_topic_name: new_topic_name ?? old_topic_name,
+                      }
+                    : undefined,
```

Comment:
Go ahead and construct this just before the big call as `const toast_params = `... giving it a name improves readability.